pyproject.toml

```diff
@@ -43,15 +43,16 @@ dependencies = [
 [project.optional-dependencies]
 lint = [
     "black==22.6.0",
-    "isort==5.10.1",
+    "isort==5.11.5",
     "flake8==3.9.2",
+    "mypy==1.8.0",
+    "typing-extensions==4.9.0",
+    "pre-commit==3.5.0",
 ]
 test = [
     "pytest-cov==3.0.0",
-    "pytest-mock==3.8.2",
     "pytest-django==4.5.2",
     "pytest==7.1.2",
-    "wagtail-factories==2.1.0",
     "dj-inmemorystorage==2.1.0"
 ]
 documentation = [
```

Comment:
Since we're not generating any MkDocs generation at the moment, and as these dependencies are rather huge compared to the other ones, I would be in favour of getting rid of that `documentation` dependencies group for now?